util-scripts/external-entities/external-entities.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+import requests
+import argparse
+import os
+import sys
+import json
+
+import tabulate
+
+
+ROX_ENDPOINT_ENV_KEY = "ROX_ENDPOINT"
+ROX_API_TOKEN_ENV_KEY = "ROX_API_TOKEN"
+
+
+def err(msg):
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+
+class Auth:
+    def __init__(self, endpoint, api_key):
+        self.endpoint = endpoint
+        self.api_key = api_key
+
+
+class Client:
+    def __init__(self, cluster, auth):
+        self._auth = auth
+        self._cluster = cluster
+        self._cluster_id = None
+
+    def get_all_external_entities(self, learned=True, cidr=None):
+        query = f"Learned External Source:{str(learned).lower()}"
+        if cidr:
+            query = f"{query}+External Source Address:{cidr}"
+        return self._get(f'v1/networkgraph/cluster/{self.cluster_id()}/externalentities', params={
+            "query": query
+        })
+
+    def get_external_flows_by_deployment(self, deployment_id, ingress_only=False, egress_only=False):
+        return self._get(f'v1/networkgraph/cluster/{self.cluster_id()}/externalentities/flows/{deployment_id}', params={
+            "ingress_only": ingress_only,
+            "egress_only": egress_only,
+        })
+
+    def get_deployment_id(self, deployment_name):
+        response = self._get('v1/deployments')
+
+        deployments = list(filter(lambda x: x['name'] == deployment_name, response.get('deployments', [])))
+        if len(deployments) != 1:
+            err(f"Unable to find deployment with name: {deployment_name}")
+
+        return deployments[0]['id']
+
+    def get_cluster_id(self):
+        response = self._get('v1/clusters', params={
+            'query': f'Cluster:{self._cluster}'
+        })
+
+        clusters = response.get('clusters', [])
+        if len(clusters) != 1:
+            err(f"Unable to find cluster with name: {self._cluster}")
+
+        cluster = clusters[0]
+        return cluster["id"]
+
+    def cluster_id(self):
+        if self._cluster_id is None:
+            self._cluster_id = self.get_cluster_id()
+        return self._cluster_id
+
+    def _get(self, path, headers=None, params=None):
+        if not headers:
+            headers = {}
+
+        headers.update({
+            'Authorization': f'Bearer {self._auth.api_key}',
+        })
+
+        response = requests.get(self._endpoint_path(path), params=params, headers=headers, verify=False).json()
+        self._handle_error_response(response)
+        return response
+
+    def _endpoint_path(self, path):
+        return f'https://{self._auth.endpoint}/{path}'
+
+    def _handle_error_response(self, response):
+        if 'error' not in response:
```

Comment:
Does this imply the API might return 200 with an error? Why are we not checking the status code?

Reply:
Ya know what I didn't even think to look at the status code, I just saw it was returning json and left it at that. I'll check and fix.